### ISSUE TITLE
Support predefined properties for all data types

### ISSUE THREAD

**Issue by bjorn** [bjorn/tiled#1410]
Apart from pre-defining properties based on object type, it would be nice to support pre-defining the properties for each data type. So defining which custom properties are valid for maps, tilesets, layers, etc.

Also [requested on the forum](http://discourse.mapeditor.org/t/layer-types-and-default-properties/1875/1).

**Comment by Yanrishatum** [comment]
Adding default properties to objects based on their tile from tileset would be really great. Our team don't use object types extensively, instad we use tile objects, and either assigning a type to them or remembering which properties they should have is not the best. :) It's way easier to just have default property list on object based on it's tile.

**Comment by bjorn** [comment]
@Yanrishatum I totally agree, but that's different from the functionality in this issue. It is covered by issues #1248 and #436.

**Comment by bjorn** [comment]
Requested on the forum [another time](http://discourse.mapeditor.org/t/is-there-a-way-to-define-a-list-of-custom-properties-for-all-tiles-in-a-tileset-instead-of-one-by-one/1940/2). In addition, there's the suggestion to add a tabular view where these predefined properties are editable in columns.

**Comment by bjorn** [comment]
[Another forum post](http://discourse.mapeditor.org/t/is-there-any-such-thing-as-layers-default-properties/2122) about a use-case that should be covered by this feature.

**Comment by bjorn** [comment]
This is now partly possible by specifying the Class of the data type. The remaining task might be to add the ability to assign a default class to each data type on the project level.

**Comment by eishiya** [comment]
> The remaining task might be to add the ability to assign a default class to each data type on the project level.

It would be very handy is this extended to more than just automatically assigned classes, but allowed custom defaults for built-in properties too, such as Object Render Order on Object Layers, Opacity and Tint Color on Layers (by layer type!), Image on Image Layers that sort of thing.

**Comment by UliAbo** [comment]
I guess also this task would be a nice addon, so it would all make sense together to have also "comments" for the properties?
https://github.com/mapeditor/tiled/issues/2133

**Comment by iocutus** [comment]
Is this still a planned feature? I would really love to describe default properties on a Tileset basis, instead of adding them to each tile manually.

**Comment by bjorn** [comment]
> Is this still a planned feature? I would really love to describe default properties on a Tileset basis, instead of adding them to each tile manually.

Yes, it is still planned, but it's not about inheriting properties from a tileset to its tiles. What you could do with this feature is to define a standard set of tile properties, which will be shown in the Properties view when a tile is selected, to make it easier to set these properties.

Note that what you can do already, it so define a class with your tile properties, then select all tiles in the tileset, and set their class to that class. This way, the class members will show as a properties on all your tiles, and if you add more members to the class they will automatically show up in the Properties view for your tiles.
